### PR TITLE
WIP: DefaultArgParser: add --format-str argument

### PR DESCRIPTION
**Pull request by meetmangukiya**
Fixes https://github.com/coala/coala/issues/2896

### CODE REVIEW

File: coalib/parsing/DefaultArgParser.py [hunk 1]
```diff
@@ -112,6 +112,15 @@ def default_arg_parser(formatter_class=None):
         '--flush-cache', const=True, action='store_const',
         help='rebuild the file cache')
 
+    if parser_type == "coala-format":
+        config_group.add_argument(
+            '--format-str', const=True, action='store',
+            help='To format the printed text by giving one or more of the '
+                'following: \n'
+                '[id, origin, column, end_column, file, line, end_line,
```
Comment (AbdealiJK) [inline coalib/parsing/DefaultArgParser.py]: need end string quote here

File: coalib/parsing/DefaultArgParser.py [hunk 2]
```diff
@@ -115,6 +115,16 @@ def default_arg_parser(formatter_class=None):
         '--flush-cache', const=True, action='store_const',
         help='rebuild the file cache')
 
+    if parser_type == "coala-format":
+        config_group.add_argument(
+            '--format-str', const=True, action='store', nargs='+',
```
Comment (sils) [inline coalib/parsing/DefaultArgParser.py]: I think this has to be store_const

File: coalib/parsing/DefaultArgParser.py [hunk 3]
```diff
@@ -115,6 +115,16 @@ def default_arg_parser(formatter_class=None):
...
+            metavar='FIELD',
```
Comment (sils) [inline coalib/parsing/DefaultArgParser.py]: why adding a metavar? We don't have this in the other places

File: coalib/parsing/DefaultArgParser.py [hunk 4]
```diff
@@ -115,6 +115,16 @@ def default_arg_parser(formatter_class=None):
...
+            help='To format the printed text by giving one or more of the '
```
Comment (sils) [inline coalib/parsing/DefaultArgParser.py]: shorten the message and start lower case, read through the others. Help texts have to be very consistent and compressed to the most necessary info only so people can read them quickly

**Comment by sils** [comment]
aaaand please fix your testcase. This is super important and I'm hapy to see it happening!

**Comment by meetmangukiya** [comment]
Preetified error(for reference) : 

```
AssertionError: 
    Regex didn't match: 'message:This file has [0-9]+ lines.' 
    not found in 
    '\x1b[31m[ERROR]\x1b[0m[05:07:13] An unknown error occurred. This is a bug. We are sorry for the inconvenience. Please contact the developers for assistance. During execution of coala an exception was raised. This should never happen. When asked for, the following information may help investigating:\x1b[0m\n\x1b[34m[INFO]\x1b[0m[05:07:13]

    Exception was:
    Traceback (most recent call last):

    file "/home/ubuntu/coala/coalib/coala_main.py", line 75, in run_coala
    arg_list=arg_list)\n\n  

    File "/home/ubuntu/coala/coalib/settings/ConfigurationGathering.py", line 353, in gather_configuration\n    
        sections, targets = load_configuration(arg_list, log_printer, arg_parser)\n\n

    File "/home/ubuntu/coala/coalib/settings/ConfigurationGathering.py", line 136, in load_configuration\n
        cli_sections = parse_cli(arg_list=arg_list, arg_parser=arg_parser)\n\n  

    File "/home/ubuntu/coala/coalib/parsing/CliParsing.py", line 37, in parse_cli\n    
        arg_parser = default_arg_parser() if arg_parser is None else arg_parser\n\n  

    File "/home/ubuntu/coala/coalib/parsing/DefaultArgParser.py", line 122, in default_arg_parser\n
        help=\'To format the printed text by giving one or more of the \'\n\n

    File "/home/ubuntu/.pyenv/versions/3.5.1/lib/python3.5/argparse.py", line 1330, in add_argument\n
        action = action_class(**kwargs)\n\n  

    File "/home/ubuntu/.pyenv/versions/3.5.1/lib/python3.5/argparse.py", line 843, in __init__\n    
        raise ValueError(\'nargs must be %r to supply const\' % OPTIONAL)\n\nValueError: nargs must be \'?\' to supply const\x1b[0m\n' : coala-format output for line count should not be empty

```

**Comment by sils** [comment]
We have a `--format` argument now that should be used for this

**Comment by meetmangukiya** [comment]
So this means, that the issue and the PR has to be closed?

**Comment by sils** [comment]
yes I think so

**Pull request closed without merging**